lib/package.js

```diff
@@ -471,9 +471,21 @@ exports.download = function(pkg, options, installDeps) {
 
         // get the override
         if (endpoint.getOverride) {
-          if (!override && config.pjson.overrides[pkg.exactName])
-            ui.log('warn', 'Using local override for `' + pkg.exactName + '`');
-          return endpoint.getOverride(pkg.registry, pkg.package, pkg.version, override || config.pjson.overrides[pkg.exactName]);
+          if (!override) {
+            for (var overrideName in config.pjson.overrides) {
+              if (overrideName.indexOf(pkg.name) !== -1) {
+                var overrideVersion = overrideName.split('@')[1];
```

Comment:
A straightforward split is not enough because `@` is an allowed character in package names. So we need to do a `lastIndexOf('@')` substr thing here.